src/main/java/fakebot/FakeBot.java

```diff
@@ -0,0 +1,311 @@
+        if (firstSplit < 0) {
+            throw new CommandException(" OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+
+        String commandName = command.substring(0, firstSplit);
+        String description = command.substring(firstSplit + 1);
+
+        if (commandName.equals(DONE_COMMAND) || commandName.equals(DELETE_COMMAND) || commandName.equals(TODO_COMMAND)
+                || commandName.equals(EVENT_COMMAND) || commandName.equals(DEADLINE_COMMAND)
+                || commandName.equals(FIND_COMMAND)) {
+            if (description.isEmpty()) {
+                throw new CommandException("☹ OOPS!!! The description of a " + commandName + " cannot be empty.");
+            } else if (commandName.equals(DONE_COMMAND) || commandName.equals(DELETE_COMMAND)) {
+                try {
+                    int index = Integer.parseInt(description);
+                    if (index > taskList.getSize() || index < 1) {
+                        throw new CommandException("☹ OOPS!!! Task number out of range.");
+                    }
+                } catch (NumberFormatException e) {
+                    throw new CommandException("☹ OOPS!!! Invalid Task Index Format.");
+                }
+                if (commandName.equals(DONE_COMMAND)) {
+                    return new Command(CommandType.DONE, description);
+
+                } else if (commandName.equals(DELETE_COMMAND)) {
+                    return new Command(CommandType.DELETE, description);
+                }
+            } else if (commandName.equals(FIND_COMMAND)) {
+                return new Command(CommandType.FIND, description);
+            } else if (commandName.equals(TODO_COMMAND)) {
+                return new Command(CommandType.TODO, description);
+            } else if (commandName.equals(DEADLINE_COMMAND)) {
+                if (!description.contains(DEADLINE_SPLIT_REGEX))
+                    throw new CommandException("☹ OOPS!!! The description of a "
+                            + DEADLINE_COMMAND + " must contain Date indicated by \""
+                            + DEADLINE_SPLIT_REGEX + "\".");
+
+                try {
+                    String[] deadlineDetails = description.split(DEADLINE_SPLIT_REGEX);
+                    String[] dates = deadlineDetails[1].split(" ");
+                    LocalDate date = LocalDate.parse(dates[0]);
+                    LocalTime time = LocalTime.parse(dates[1]);
+                } catch (Exception e) {
+                    throw new CommandException("☹ OOPS!!! The Date format of a "
+                            + DEADLINE_COMMAND + " must be yyyy-mm-dd hh:ss.");
+
+                }
+
+                return new Command(CommandType.DEADLINE, description);
+            } else if (commandName.equals(EVENT_COMMAND)) {
+                if (!description.contains(EVENT_SPLIT_REGEX))
+                    throw new CommandException("☹ OOPS!!! The description of a " + EVENT_COMMAND
```

Comment:
I like how you handled the exceptions for a lot of different cases! (:

Reply:
thank you